Purchases/SwiftSources/IntroEligibilityCalculator.swift

```diff
@@ -36,20 +36,26 @@ public class IntroEligibilityCalculator: NSObject {
         var result: [String: Int] = candidateProductIdentifiers.reduce(into: [:]) { resultDict, productId in
             resultDict[productId] = IntroEligibilityStatus.unknown.rawValue
         }
-        
-        let purchasedProductIdsWithIntroOffers = localReceiptParser.purchasedIntroOfferProductIdentifiers(receiptData: receiptData)
-        
-        let allProductIdentifiers = candidateProductIdentifiers.union(purchasedProductIdsWithIntroOffers)
-        
-        productsManager.products(withIdentifiers: allProductIdentifiers) { allProducts in
-            let purchasedProductsWithIntroOffers = allProducts.filter { purchasedProductIdsWithIntroOffers.contains($0.productIdentifier) }
-            let candidateProducts = allProducts.filter { candidateProductIdentifiers.contains($0.productIdentifier) }
+        do {
```

Comment:
the code here is the same, just in a `do try catch` block, since our code can throw if something goes wrong when parsing the receipt.